### ISSUE TITLE
Feature request: recursive `put`

### ISSUE THREAD

**Issue by nicholasbishop** [AmadeusITGroup/JumpSSH#25]
Hope it's OK to request a feature here, didn't see anything in the documentation about that.

Feature request is a recursive version of `SSHSession.put` for uploading an entire directory to the remote.

**Comment by t-cas** [comment]
Yes, perfectly fine to request features here, I will add a note about that in the documentation.
Contributions are also welcome :-)
Thanks for this request, recursive put is indeed something that could be really useful.
